2 Niagara Fundamentals Exercise - Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiPlexN.java

```diff
@@ -0,0 +1,168 @@
+import javax.baja.nre.annotations.NiagaraType;
+import javax.baja.sys.*;
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+import java.util.List;
+
+@NiagaraType
+@NiagaraProperty(
+        name = "out",
+        type = "BBoolean",
+        defaultValue = "BBoolean.make(false)",
+        flags = Flags.READONLY | Flags.SUMMARY | Flags.DEFAULT_ON_CLONE
+)
+@NiagaraAction(
+        name = "addInput"
+)
+@NiagaraAction(
+        name = "addSelector"
+)
+//TO-DO delete Action for both types
+public class BMultiPlexN extends BComponent
+{
+//region /*+ ------------ BEGIN BAJA AUTO GENERATED CODE ------------ +*/
+//@formatter:off
+/*@ $com.tridiumuniversity.multiplexer.BMultiPlexN(2303573508)1.0$ @*/
+/* Generated Wed Aug 21 22:45:00 CEST 2024 by Slot-o-Matic (c) Tridium, Inc. 2012-2024 */
+
+  //region Property "out"
+
+  /**
+   * Slot for the {@code out} property.
+   * @see #getOut
+   * @see #setOut
+   */
+  public static final Property out = newProperty(Flags.READONLY | Flags.SUMMARY | Flags.DEFAULT_ON_CLONE, BBoolean.make(false).as(BBoolean.class).getBoolean(), null);
+
+  /**
+   * Get the {@code out} property.
+   * @see #out
+   */
+  public boolean getOut() { return getBoolean(out); }
+
+  /**
+   * Set the {@code out} property.
+   * @see #out
+   */
+  public void setOut(boolean v) { setBoolean(out, v, null); }
+
+  //endregion Property "out"
+
+  //region Action "addInput"
+
+  /**
+   * Slot for the {@code addInput} action.
+   * @see #addInput()
+   */
+  public static final Action addInput = newAction(0, null);
+
+  /**
+   * Invoke the {@code addInput} action.
+   * @see #addInput
+   */
+  public void addInput() { invoke(addInput, null, null); }
+
+  //endregion Action "addInput"
+
+  //region Action "addSelector"
+
+  /**
+   * Slot for the {@code addSelector} action.
+   * @see #addSelector()
+   */
+  public static final Action addSelector = newAction(0, null);
+
+  /**
+   * Invoke the {@code addSelector} action.
+   * @see #addSelector
+   */
+  public void addSelector() { invoke(addSelector, null, null); }
+
+  //endregion Action "addSelector"
+
+  //region Type
+
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BMultiPlexN.class);
+
+  //endregion Type
+
+//@formatter:on
+//endregion /*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+  @Override
+  public void changed(Property p, Context cx)
+  {
+    if (!isRunning()) return;
+    calculate();
+  }
+  @Override
+  public void added(Property p, Context cx)
+  {
+    countDynamicsSlots();
+  }
+  @Override
+  public void removed(Property property, BValue oldValue, Context context)
+  {
+    countDynamicsSlots();
+  }
+  public void doAddInput()
+  {
+    //To-do maximum number of input slots
+    int flags = Flags.SUMMARY | Flags.DEFAULT_ON_CLONE;
+    add("in0?", BBoolean.make(false).as(BBoolean.class), flags); // BBoolean.make(false).as(BBoolean.class).getBoolean()
```

Comment:
I don't think you need the `as(BBoolean.class)` call here. In fact, we could simplify this line to:

```java
add("in0?", BBoolean.FALSE, flags)
```